src/ensure_plugin/java/me/liliandev/ensure/ensures/EnsureNotNull.java

```diff
@@ -0,0 +1,16 @@
+package me.liliandev.ensure.ensures;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.PARAMETER})
+/**
+ * Adds a null check to non primitive parameters
+ */
+public @interface EnsureNotNull {
```

Comment:
This isn't currently used right?

Reply:
yes, but I had a few ideas where it makes sense so I added it without checking all potential spots and using it atm

Reply:
Yeah makes sense